pkg/staticnetworkconfig/generator.go

```
@@ -71,8 +71,15 @@ func (s *StaticNetworkConfigGenerator) injectIPV4V6FieldsIfNeeded(networkYaml st
 		s.log.WithError(err).Errorf("Error unmarshalling yaml")
 		return "", err
 	}
-	Interfaces := config["interfaces"].([]interface{})
-	for _, iface := range Interfaces {
+	interfaces, exists := config["interfaces"]
+	if !exists || interfaces == nil {
+		return "", nil
+	}
+	interfacesSlice, ok := interfaces.([]interface{})
+	if !ok {
+		return "", nil
```

don't we want to return an error here?